### PR TITLE
Performance

### PR DESCRIPTION
**Pull request by Neeraj-Nec**
incorporated comment of document and change the code of broker to handle the cached notification

### CODE REVIEW

**Review by showersky — APPROVED** [review]

File: doc/en/source/ngsildbasediotintegration.rst [hunk 1]
```diff
@@ -220,30 +220,7 @@ communications protocol to be used.
...
-
-
-**step 4** To see the status of entity **urn:ngsi-ld:Device:water001** open the device dashboard in your web browser by using URL: **<IoT-DeviceIP>:3000/device/monitor** . The status should be "on".
-
-.. figure:: figures/status1.png
-
-
-
-**step 5** To verify the result from orion broker send the following request
+**step 3** After Provisioning an Actuator on IoT agent the IoT agent record a measurement of Actuator on orion-broker. To retrive the recorded measurement of actuater from context broker execute the following command.
```
Comment (naveennec) [inline doc/en/source/ngsildbasediotintegration.rst:223]: correct the step "IoT Agent records the measurement of Actuator after Provisioning the Actuator on it  and forward the measurement to Orion. Execute the following command to retrieve the recorded measurement of actuator from Orion"

File: doc/en/source/ngsildbasediotintegration.rst [hunk 2]
```diff
@@ -339,13 +311,36 @@ Response
 		}
 	}
 	
-The **observedAt** shows last the time any command associated with the entity has been invoked. The result of **on** command can be seen in the value of the **on_info** attribute.
+
+**Step 4** To see the state of the water sprinkler change through device monitor URL:**<IoT-DeviceIP>:3000/device/monitor** send the below PATCH request directly to the IoT Agent's North Port
```
Comment (naveennec) [inline doc/en/source/ngsildbasediotintegration.rst:315]: Better to write "To observe"

**Review by naveennec — CHANGES_REQUESTED** [review]

File: broker/thinBroker.go [hunk 3]
```diff
@@ -2909,24 +2933,27 @@ func (tb *ThinBroker) sendReliableNotifyToNgsiLDSubscriber(elements []map[string
...
 
 		tb.ldSubscriptions_lock.Lock()
 		if ldSubscription, exist := tb.ldSubscriptions[sid]; exist {
+			fmt.Println("ldsubscription", ldSubscription)
+			fmt.Println("reliabolity", ldSubscription.Subscriber.RequireReliability)
 			if ldSubscription.Subscriber.RequireReliability == true {
 				ldSubscription.Subscriber.LDNotifyCache = append(ldSubscription.Subscriber.LDNotifyCache, elements...)
-				ldSubscription.Notification.LastFailure = notifyTime
-				ldSubscription.Notification.Status = "failed"
+				fmt.Println("")
```
[the remark this answers is not in the document]
Comment (Neeraj-Nec) [reply broker/thinBroker.go:2947]: resolved

File: broker/thinBroker.go [hunk 4]
```diff
@@ -2909,24 +2933,27 @@ func (tb *ThinBroker) sendReliableNotifyToNgsiLDSubscriber(elements []map[string
...
+				//ldSubscription.Notification.LastFailure = notifyTime
```
[the remark this answers is not in the document]
Comment (Neeraj-Nec) [reply broker/thinBroker.go:2948]: resolved

File: doc/en/source/ngsildbasediotintegration.rst (hunk 1 above)
Comment (Neeraj-Nec) [reply doc/en/source/ngsildbasediotintegration.rst]: Done

**Review by naveennec — APPROVED** [review]

**Pull request merged**